Basilisk master bc2 bjs2 1884

TrueSamedi opened the pull request.

TrueSamedi committed:
add PostDto
add PostService
add ScheduledPostPublisher
fixed post_V003__resource.sql

TrueSamedi committed:
add ThreadPoolConfig
add PostServiceTest
fixed PostService
fixed ScheduledPostPublisher

TrueSamedi committed:
fixed ThreadPoolConfig
fixed PostServiceTest
fixed PostService

TrueSamedi committed:
fixed PostServiceTest
fixed PostService

PavelAP96 commented on src/main/java/faang/school/postservice/publisher/ScheduledPostPublisher.java:
```diff
@@ -0,0 +1,19 @@
+
+    private final PostService postService;
+
+    @Scheduled(fixedRate = 60000)
```
надо вынести этот параметр в application.yaml, чтобы он был настраиваемым

PavelAP96 commented on src/main/java/faang/school/postservice/service/PostService.java:
```diff
@@ -0,0 +1,44 @@
+        }
+    }
+
+    @Async(value = "poolForScheduled")
```
А ты уверен, что выполняется асинхронно?
как будто бы не хватает аннотации EnableAsync

PavelAP96 replied:
Еще есть ощущение, что не асинхронно, потому что
Асинхронные методы должны вызываться из другого компонента. То есть, если мы вызываем метод с аннотацией @Async из того же класса, где он определен, то это не будет работать;

PavelAP96 commented on src/main/java/faang/school/postservice/config/ThreadPoolConfig.java:
```diff
@@ -0,0 +1,17 @@
+@Configuration
+public class ThreadPoolConfig {
+
+    @Bean(name = "poolForScheduled")
```
название чет не очень для бина
назови так, чтобы было было понятно для какого события этот пул
типа postPublishingThreadPool
и старайся не использовать в названиях слова типа for, используй их когда без них совсем никак

PavelAP96 commented on src/main/java/faang/school/postservice/service/PostService.java:
```diff
@@ -0,0 +1,44 @@
+    public void publishScheduledPosts() {
+        List<Post> posts = postRepository.findReadyToPublish();
+
+        for (int i = 0; i < posts.size(); i += 1000) {
```
Параметр batchSize должен быть настраиваемым
у тебя это всегда 1000, а хотелось бы , чтобы это можно было настраваить, вынеси его в appliatiuon.yaml

PavelAP96 commented on src/main/java/faang/school/postservice/service/PostService.java:
```diff
@@ -0,0 +1,44 @@
+        List<Post> posts = postRepository.findReadyToPublish();
+
+        for (int i = 0; i < posts.size(); i += 1000) {
+            int finalI = i;
+            if (i + 1000 < posts.size()) {
+                publishThousandPosts(posts, finalI, finalI + 1000);
+            } else {
+                publishThousandPosts(posts, finalI, posts.size());
+            }
```
чтобы не париться с этой всей логикой и не писать ее вручную, посмотри в сторону ListUtils.partition

TrueSamedi replied:
мне реально проще писать и читать через циклы)) так намного прозрачнее))

TrueSamedi committed:
add AsyncPostService
fixed application.yaml
fixed PostService
fixed ScheduledPostPublisher
fixed ThreadPoolConfig

PavelAP96 committed:
Merge branch 'basilisk-master-bc2' into basilisk-master-bc2-BJS2-1884
